graphbolt/src/cuda/neighbor_sampler.cu

```diff
@@ -528,18 +534,107 @@ c10::intrusive_ptr<sampling::FusedSampledSubgraph> SampleNeighbors(
         }
       }));
 
-  // Convert output_indptr back to homo by discarding intermediate offsets.
-  output_indptr =
-      output_indptr.slice(0, 0, output_indptr.size(0), fanouts.size());
+  torch::optional<torch::Tensor> edge_offsets = torch::nullopt;
+  if (type_per_edge) {
+    const int64_t num_etypes =
+        edge_type_to_id.has_value() ? edge_type_to_id->size() : 1;
+    if (fanouts.size() == 1) {
+      torch::Tensor output_in_degree, sliced_output_indptr;
+      sliced_output_indptr =
+          output_indptr.slice(0, 0, output_indptr.size(0) - 1);
```

Comment:
What does this slice do? just get rid of the last element?

Reply:
Yes.